terragrunt/aws/config/wafv2_logs.tf

```diff
@@ -0,0 +1,243 @@
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_lambda_permission" "cbs_wafv2_logs_rule" {
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.cbs_wafv2_logs_rule.arn
+  principal     = "config.amazonaws.com"
+  statement_id  = "AllowExecutionFromConfig"
+}
+
+resource "aws_cloudwatch_log_group" "cbs_wafv2_logs_rule" {
+  name              = "/aws/lambda/${aws_lambda_function.cbs_wafv2_logs_rule.function_name}"
+  retention_in_days = 14
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+#
+# Lambda execution role
+#
+resource "aws_iam_role" "cbs_wafv2_logs_rule" {
+  name               = "cbs_wafv2_logs_rule"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume.json
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "cbs_wafv2_logs_rule_basic_execution" {
+  role       = aws_iam_role.cbs_wafv2_logs_rule.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "cbs_wafv2_logs_rule_wafv2_list_web_acls" {
+  role       = aws_iam_role.cbs_wafv2_logs_rule.name
+  policy_arn = aws_iam_policy.wafv2_list_web_acls.arn
+}
+
+resource "aws_iam_policy" "wafv2_list_web_acls" {
+  name        = "wafv2_list_web_acls"
+  path        = "/"
+  description = "IAM policy for listing all WAFv2 ACLs"
+  policy      = data.aws_iam_policy_document.wafv2_list_web_acls.json
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+data "aws_iam_policy_document" "wafv2_list_web_acls" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "wafv2:GetWebACL",
+      "wafv2:GetLoggingConfiguration",
+      "wafv2:ListWebACLs",
+      "wafv2:ListLoggingConfigurations",
+      "wafv2:PutLoggingConfiguration",
+      "config:PutEvaluations",
+      "iam:CreateServiceLinkedRole",
+    ]
+    resources = ["*"]
+  }
+}
+
+resource "aws_cloudwatch_log_group" "cbs_default_kinesis_stream" {
+  name              = "/aws/kinesisfirehose/cbs_default_kinesis_stream"
+  retention_in_days = 14
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "cbs_default_stream" {
+  name        = "aws-waf-logs-${var.account_id}"
+  destination = "extended_s3"
+
+  extended_s3_configuration {
+    role_arn   = aws_iam_role.waf_log_role.arn
+    prefix     = "waf/${var.account_id}/"
+    bucket_arn = module.waf_logs.s3_bucket_arn
+    cloudwatch_logging_options {
+      enabled         = true
+      log_group_name  = aws_cloudwatch_log_group.cbs_default_kinesis_stream.name
+      log_stream_name = "WAFLogS3Delivery"
+    }
+  }
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+module "waf_logs" {
```

Comment:
I'm assuming we'll want a replication rule on this resource that sends the logs to the satellite s3 bucket?

Reply:
@patheard i was curious about this. Will your rule automatically do this ? I didn't want to duplicate aws config rules so i removed the replication to the satellite bucket.

Reply:
My thinking had been that there's a lot of config and IAM involved in replicating cross-account with a CMK, so I wanted to limit that setup to only one S3 bucket.

Reply:
refactored to replicate to the satellite bucket.